src/Engine/ProtoCore/RuntimeStatus.cs

```diff
@@ -371,32 +383,47 @@ private void GetLocationByGraphNode(ref int line, ref int col)
             }
             else if (CoreUtils.TryGetOperator(methodName, out op))
             {
-                string strOp = Op.GetOpSymbol(op);
+                var strOp = Op.GetOpSymbol(op);
                 message = String.Format(Resources.kMethodResolutionFailureForOperator,
                                         strOp,
-                                        runtimeCore.DSExecutable.TypeSystem.GetType(arguments[0].metaData.type),
-                                        runtimeCore.DSExecutable.TypeSystem.GetType(arguments[1].metaData.type));
+                                        GetTypeName(arguments[0]),
+                                        GetTypeName(arguments[1]));
             }
-            else
+            else if (funcGroup.FunctionEndPoints.Count == 1) // non-overloaded case
```

Comment:
Ok, just checked, the number of function endpoints can still be one if there are function overloads but with different number of arguments.